spec/app/ics-027-interchain-accounts/README.md

```diff
@@ -16,114 +16,121 @@ This standard document specifies packet data structure, state machine handling l
 
-Each chain must satisfy following features to create a interchain account:
+### Authentication & Authorization
 
-- New interchain accounts must not conflict with existing ones.
-- Each chain must keep track of which counterparty chain created each new interchain account.
+For the controller chain to register an interchain account on a host chain, first the controlling side must bind a new port to the interchain account module with the id `ics27-1-{connection-number}-{owner-address}` where `{owner-address}` is the account address of the interchain account owner & connection number is the number parsed from the connection id (e.g. `connection-1` would have a connection number of `1`). This port will be used to create channels between the controller & host chain for a specific owner/interchain account pair. Only the account with `{owner-address}` matching the bound port will be authorized to send IBC packets over channels created with `SourcePort` `ics27-1-{connection-number}-{owner-address}`. The host chain trusts that each controller chain will enforce this port registration and access. 
 
-Also, each chain must know how the counterparty chains serialise/deserialise transaction bytes in order to send transactions via IBC. And the counterparty chain must implement the process of safely executing IBC transactions by verifying the authority of the transaction's signers.
 
-The chain must reject the transaction and must not make a state transition in the following cases:
+The controller and host chain should keep track of an `active-channel` for each registered interchain account. The `active-channel` is set during the channel creation handshake process. If a channel closes, the `active-channel` should be unset. A new channel can be opened with the same source port, allowing access to the interchain account on the host chain. 
 
-- The IBC transaction fails to be deserialised.
-- The IBC transaction expects signers other than the interchain accounts made by the counterparty chain.
+An active channel can look like this:
 
-It does not restrict how you can distinguish signers that was not made by the counterparty chain. But the most common way would be to record the account in state when the interchain account is registered and to verify that signers are recorded interchain account.
+
+```typescript
+{
+ // Controller Chain
+ SourcePortId: `ics-27-1-cosmos1mjk79fjjgpplak5wq838w0yd982gzkyfrk07am`,
+ SourceChannelId: `channel-1`,
+ // Host Chain
+ CounterPartyPortId: `interchain_account`,
+ CounterPartyChannelId: `channel-2`,
+}
+```
 
 ### Data Structures
 
-Each chain must implement the interfaces as defined in this section in order to support interchain accounts.
+```typescript
+interface InterchainAccountModule {
```

Comment:
Well, the helper module uses the controller part of the `InterchainAccountModule` API. I wouldn't necessarily say that the helper module is the controller module per say (although in some ways it is). 

I think for now a single interface is fine, but in terms of readability of code, it may be nice to have two separate interfaces in the future.